cmd/api/src/api/v2/auth/oidc.go

```diff
@@ -207,7 +231,44 @@ func (s ManagementResource) OIDCCallbackHandler(response http.ResponseWriter, re
 				// Technical or credentials issue
 				// Not explicitly covered; treat as a technical issue
 				redirectToLoginPage(response, request, "We’re having trouble connecting. Please check your internet and try again.")
+			} else if claims.Email == "" {
+				api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, "email claim is missing", request), response)
+				return
 			} else {
+				if ssoProvider.Config.AutoProvision.Enabled {
+					if user, err := s.db.LookupUser(request.Context(), claims.Email); err != nil {
+						if errors.Is(err, database.ErrNotFound) {
```

Comment:
The `errors.Is` can be hoisted up as the first check instead of `err != nil`, as it's safe against nil. This would also make it clearer that we missed the default error handling, which otherwise would go unnoticed here. The final form would be something like:

```go
if user, err := s.db.LookupUser(request.Context(), claims.Email); errors.Is(err, database.ErrNotFound) {
// Role stuff goes here
} else if err != nil {
// missing error handling goes here
}
```

There may be other chances to refactor to avoid so much nested checking, and I'd encourage further tinkering to avoid unnecessarily nested code to keep your happy path left aligned as much as possible